frontend/packages/pipelines-plugin/src/components/repository/list-page/RepositoryRow.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { TableRow, TableData, RowFunction } from '@console/internal/components/factory';
+import {
+  Kebab,
+  LoadingInline,
+  ResourceKebab,
+  ResourceLink,
+  Timestamp,
+} from '@console/internal/components/utils';
+import { referenceFor, referenceForModel } from '@console/internal/module/k8s';
+import { PipelineRunModel, RepositoryModel } from '../../../models';
+import { pipelineRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
+import { pipelineRunDuration } from '../../../utils/pipeline-utils';
+import LinkedPipelineRunTaskStatus from '../../pipelineruns/status/LinkedPipelineRunTaskStatus';
+import PipelineRunStatus from '../../pipelineruns/status/PipelineRunStatus';
+import { useGetLastPLR } from '../repository-hooks';
+import { RepositoryKind } from '../types';
+import { repositoriesTableColumnClasses } from './RepositoryHeader';
+
+type RepositoryRowWrapperProps = {
+  obj: RepositoryKind;
+  index: number;
+  key: string;
+  style: object;
+};
+
+export const RepositoryRowWrapper: React.FC<RepositoryRowWrapperProps> = ({
+  obj,
+  index,
+  key,
+  style,
+}) => {
+  const {
+    metadata: { name, namespace },
+  } = obj;
+  const pipelineRunsName = React.useMemo(
+    () => obj.pipelinerun_status?.map((plrStatus) => plrStatus.pipelineRunName),
+    [obj],
+  );
```

Comment:
Instead of fetching all the pipelineRuns in pipelinerun_status to find the latestRun, I think pipelinerun_status will already have `completionTime` and `startTime` we could rely on completionTime (fallback is to use startTime) and find the latest pipelineRunName and make only one call to find all other details like status. This would reduce the load time when there are more Repositories IMO.
```suggestion
    const pipelineRunsName = React.useMemo(() => {
    const runNames = obj.pipelinerun_status
      ?.sort((a, b) => {
        if (a.completionTime) {
          return b?.completionTime && new Date(a.completionTime) > new Date(b.completionTime)
            ? 1
            : -1;
        }
        return b?.completionTime || new Date(a?.startTime) > new Date(b.startTime) ? 1 : -1;
      })
      .map((plrStatus) => plrStatus.pipelineRunName);
    return [runNames[runNames.length - 1]];
  }, [obj]);
  );
```